script.js

```diff
@@ -2,124 +2,280 @@
 const quizArrayOfObjects = []
 
 
-// generate quiz btn true/false 
-const GenerateQuizbtn = () => {
-    const generateQuizButton = document.getElementById("generateQuiz")
-    generateQuizButton.disabled = quizArrayOfObjects.length === 0 // set true/false
-}
+// menu bar 
+const body = document.body
+// menuDiv
+const menuDiv = document.createElement("div")
+menuDiv.setAttribute("id", "menuDiv")
+body.prepend(menuDiv)
+// menuDiv -> Buttons
+const buttons = [
+    { text: "Add Quiz", id: "addQuizBtn" },
+    { text: "Quiz List", id: "quizListBtn" },
+    { text: "Add Players & Run Competition", id: "addPlayersBtn" }
+]
+buttons.forEach(({ text, id }) => {
+    const button = document.createElement("button")
+    button.setAttribute("type", "button")
+    button.setAttribute("id", id)
+    button.innerHTML = text
+    menuDiv.appendChild(button)
+})
 
 
-// validations for options and correct answer
-const ValidationAndColor=()=>{
-    // for color
-    document.addEventListener("DOMContentLoaded", () => {
-        // get inputs
-        const correctAnswer = document.querySelector("#correctAnswer")
-        const option1Input = document.querySelector("#option1")
-        const option2Input = document.querySelector("#option2")
-        const option3Input = document.querySelector("#option3")
-        const option4Input = document.querySelector("#option4")
-    
-        correctAnswer.addEventListener("input", (event) => {
-            // get values
-            const correctAnswerValue = event.target.value.trim().toLowerCase()
-            const option1 = option1Input.value.trim().toLowerCase()
-            const option2 = option2Input.value.trim().toLowerCase()
-            const option3 = option3Input.value.trim().toLowerCase()
-            const option4 = option4Input.value.trim().toLowerCase()
-    
-            //  allInputs array
-            const allInputs = [option1Input, option2Input, option3Input, option4Input]
+// create quiz object 
+const createQuizObject = (question, option1, option2, option3, option4, correctAnswer) => {
+    if (!question || !option1 || !option2 || !option3 || !option4 || !correctAnswer) {
+        return "All fields are required."
+    }
+    // quiz object
+    const quiz = { question, 
+        options: [option1, option2, option3, option4], 
+        correctAnswer }
     
-            // reset colors 
-            allInputs.forEach(input => {
-                input.style.backgroundColor = ""
+        // push
+    quizArrayOfObjects.push(quiz)
+    return quiz
+}
+// reset input colors 
+const validationResetColor = () => {
+    document.querySelectorAll("#option1, #option2, #option3, #option4").forEach((input) => {
+        input.style.backgroundColor = ""
+    })
```

Comment:
Imagine you had a class name that applies styling to all those. Then you could simply remove the class or change your `querySelectorAll` to be much more simple:
```suggestion
    document.querySelectorAll(".option-input").forEach((input) => {
    input.classList.remove('option-input--invalid');
```
or
```suggestion
    document.querySelectorAll(".option-input").forEach((input) => {
     input.style.backgroundColor = ""
```

Reply:
I got your point. Option one is the best approach to add a common class to all inputs and remove it whenever needed, especially for handling colors (**Best point and learning**). Next time i will apply this approach and do experiment like how to add class and how to remove on event.